Lib/fontTools/subset/cff.py

```diff
@@ -352,28 +352,45 @@ def processSubr(self, index, subr):
 				if subr_hints.status == 0:
 					hints.last_hint = index
 				else:
-					hints.last_hint = index - 2 # Leave the subr call in
+					hints.last_hint = index - 2  # Leave the subr call in
 
 		elif subr_hints.status == 0:
 			hints.deletions.append(index)
 
 		hints.status = max(hints.status, subr_hints.status)
 
+class StopHintCountEvent(Exception):
+	pass
+
+
+stop_hintcount_ops = ("op_hstem", "op_vstem", "op_rmoveto", "op_hmoveto",
+						"op_vmoveto")
+
+
 class _DesubroutinizingT2Decompiler(psCharStrings.SimpleT2Decompiler):
 
 	def __init__(self, localSubrs, globalSubrs, private=None):
-		psCharStrings.SimpleT2Decompiler.__init__(self,
-							  localSubrs,
-							  globalSubrs, private)
+		psCharStrings.SimpleT2Decompiler.__init__(self, localSubrs, globalSubrs,
+												private)
 
 	def execute(self, charString):
+		self.need_hintcount = True  # until proven otherwise
+		for op_name in stop_hintcount_ops:
+			setattr(self, op_name, self.stop_hint_count)
+
 		if hasattr(charString, '_desubroutinized'):
+			if self.need_hintcount and self.callingStack:
+				try:
+					psCharStrings.SimpleT2Decompiler.execute(self, charString)
+				except StopHintCountEvent:
+					del self.callingStack[-1]
+				return
```

Comment:
this `return` statement is also unnecessary, the following line does the same

Reply:
Right.